### DISCUSSION TITLE
Memory usage

### DISCUSSION THREAD

**Discussion by mammadmaheri7** [zkonduit/ezkl#768] [General]
I encountered an issue while attempting to verify the tiny_vit model, which contains approximately 5 million parameters. During the setup step, it is consuming a surprisingly large amount of RAM, exceeding even 1000GB (though the exact amount is unknown). This leads me to wonder whether there might be a bug in the process or if adjusting certain settings variables, such as `num_cols`, `logrows`,`tolerance` ,etc., could potentially resolve the problem.

The commands I executed are as follows (the onnx file, calibration data and settings after calibration have been attached in the zip file):

1. `ezkl gen-settings -M output.onnx -O settings.json --input-visibility public --param-visibility fixed --output-visibility public`
2. `ezkl calibrate-settings -D cal_data.json -M output.onnx -O settings.json --target resource --lookup-safety-margin 1 --scales 0,5,10,15 --only-range-check-rebase`
3. `ezkl compile-circuit -M output.onnx -S settings.json --compiled-circuit network.ezkl`
4. `ezkl get-srs -S settings.json --srs-path kzg.srs`
5. `ezkl setup -M network.ezkl --srs-path=kzg.srs --vk-path=vk.key --pk-path=pk.key`

Additionally, I would like to specify that I built the ezkl tool from the following commit:
[Link to Commit](https://github.com/zkonduit/ezkl/commit/ff563e93a74e88344109abea6253822351227426)

[tiny_vit.zip](https://github.com/zkonduit/ezkl/files/14875339/tiny_vit.zip)

**Comment by alexander-camuto** [comment]
This is a pretty big model in terms of num of constraints -- so I think the memory usage makes sense. I'd note that num of params is not always a good proxy for num of constraints (architectures will induce very big differences)

**Reply by alexander-camuto** [reply]
you can reduce the num of rows by playing around with the `--num-inner-cols` parameter (at the expense possibly of larger proof times)
